test/unit/std/charconv_test.cpp

```diff
@@ -0,0 +1,502 @@
+    auto res = std::from_chars(&str[0], &str[0] + str.size(), value);
+
+    EXPECT_EQ(res.ptr, &str[0] + str.size());
+    EXPECT_EQ(res.ec, std::errc::result_out_of_range);
+    EXPECT_EQ(value, TypeParam{42});
+}
+
+TYPED_TEST(integral_from_char_test, partial_parsing)
+{
+    TypeParam value{42};
+
+    { // interleaved char
+        std::vector<char> const str{'1', 'a', '3'};
+
+        auto res = std::from_chars(&str[0], &str[0] + str.size(), value);
+
+        EXPECT_EQ(res.ptr, &str[1]);
+        EXPECT_EQ(res.ec, std::errc{});
+        EXPECT_EQ(value, TypeParam{1});
+    }
+
+    value = 42; // reset
+    { // trailing char
+        std::vector<char> const str{'1', '2', 'a'};
+
+        auto res = std::from_chars(&str[0], &str[0] + str.size(), value);
+
+        EXPECT_EQ(res.ptr, &str[2]);
+        EXPECT_EQ(res.ec, std::errc{});
+        EXPECT_EQ(value, TypeParam{12});
+    }
+
+    value = 42; // reset
+    { // float
+        std::vector<char> const str{'1', '.', '3'};
+
+        auto res = std::from_chars(&str[0], &str[0] + str.size(), value);
+
+        EXPECT_EQ(res.ptr, &str[1]);
+        EXPECT_EQ(res.ec, std::errc{});
+        EXPECT_EQ(value, TypeParam{1});
+    }
+
+    { // hexadecimal 0x prefix is not recognized
+        std::vector<char> const str{'0', 'x', '3', 'f'};
+
+        auto res = std::from_chars(&str[0], &str[0] + str.size(), value, 16);
+
+        EXPECT_EQ(res.ptr, &str[1]);
+        EXPECT_EQ(res.ec, std::errc{});
+        EXPECT_EQ(value, TypeParam{0});
+    }
+
+    { // hexadecimal 0X prefix is not recognized
+        std::vector<char> const str{'0', 'X', '3', 'f'};
+
+        auto res = std::from_chars(&str[0], &str[0] + str.size(), value, 16);
+
+        EXPECT_EQ(res.ptr, &str[1]);
+        EXPECT_EQ(res.ec, std::errc{});
+        EXPECT_EQ(value, TypeParam{0});
+    }
+}
+
+TYPED_TEST(integral_from_char_test, invalid_argument_error)
+{
+    TypeParam value{42};
+
+    { // leading char
+        std::vector<char> const str{'a', '1', '3'};
+
+        auto res = std::from_chars(&str[0], &str[0] + str.size(), value);
+
+        EXPECT_EQ(res.ptr, &str[0]);
+        EXPECT_EQ(res.ec, std::errc::invalid_argument);
+        EXPECT_EQ(value, TypeParam{42});
+    }
+
+    { // leading + sign (or do we want this to succeed?)
+        std::vector<char> const str{'+', '1', '3'};
+
+        auto res = std::from_chars(&str[0], &str[0] + str.size(), value);
+
+        EXPECT_EQ(res.ptr, &str[0]);
+        EXPECT_EQ(res.ec, std::errc::invalid_argument);
+        EXPECT_EQ(value, TypeParam{42});
+    }
+
+    { // hexadecimal x prefix is not recognized
+        std::vector<char> const str{'x', '3', 'f'};
+
+        auto res = std::from_chars(&str[0], &str[0] + str.size(), value, 16);
+
+        EXPECT_EQ(res.ptr, &str[0]);
+        EXPECT_EQ(res.ec, std::errc::invalid_argument);
+        EXPECT_EQ(value, TypeParam{42});
+    }
+}
+
+// GCC and Clang have different behaviour in debug mode for an invalid base parameter
```

Comment:
Hm, if the behaviour is not specified, we shouldn't test it I think.

If you really want to leave this code inside, do not comment it via `//`, instead use `#if 0` and clearly state under which condition it shall be activated again.